client/my-sites/stats/stats-purchase/stats-pwyw-uprade-slider/index.tsx

```diff
@@ -66,25 +43,30 @@ function emojiForStep( index: number, uiEmojiHeartTier: number, uiImageCelebrati
 	return String.fromCodePoint( 0x1f525 );
 }
 
-// Takes a StatsPWYWSliderSettings object and returns an array of slider steps.
-// The slider wants string values for the left and right labels.
-function stepsFromSettings( settings: StatsPWYWSliderSettings, currencyCode: string ) {
-	// Pull tier strategy from settings.
-	// We ignore the emoji thresholds and use our own.
-	const { sliderStepPrice, minSliderPrice, maxSliderPrice } = settings;
-	// Set up our slider steps based on above strategy.
-	const sliderSteps = [];
-	const maxSliderValue = Math.floor( maxSliderPrice / sliderStepPrice );
-	const minSliderValue = Math.round( minSliderPrice / sliderStepPrice );
-	for ( let i = minSliderValue; i <= maxSliderValue; i++ ) {
-		const rawValue = minSliderPrice + i * sliderStepPrice;
-		sliderSteps.push( {
-			raw: rawValue,
-			lhValue: formatCurrency( rawValue, currencyCode ),
-			rhValue: emojiForStep( i, settings.uiEmojiHeartTier, settings.uiImageCelebrationTier ),
-		} );
+// Generate the range of available tiers based on the passed-in slider settings.
+function generatePlanTiers( settings: StatsPWYWSliderSettings ) {
```

Comment:
Do we want to reuse the `generatePlanTiers` in other places? Or could we consolidate `generateSteps` and  `generatePlanTiers` since there are no actual tiers in the PWYW purchase flow?

Reply:
I think that Dognose is right. Mixing the terminology makes it confusing.

Reply:
@a8ck3n Feel free to ship and follow up on the discussion in another PR.

Reply:
I was not planning to reuse it but I wanted to keep them separate and distinct for now as I expect the logic will need to be revisited if we allow for tier upgrades within the PWYW flow.